plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -300,13 +369,22 @@ class bp_connection_manager {
          return true;
       };
 
+      const auto head_block_time = self()->head_block_time.load();
+      const block_timestamp_type expire = head_block_time + bp_gossip_peer_expiration_variance;
+      auto is_expiration_valid = [&](const gossip_bp_peers_message::signed_bp_peer& peer) -> bool {
+         if (initial_msg)
+            return true; // initial message has no expiration
+         return peer.expiration > head_block_time && peer.expiration < expire;
+      };
```

Comment:
I don't understand this function.

I think the criteria for being `valid` should be `peer.expiration > expire`, not the opposite?

Why are we adding `bp_gossip_peer_expiration_variance` to `expire`?

Reply:
This is looking at the expiration that was communicated to us from another node. To be a valid expiration it has to be greater than the current head block time and also less than 1hr 15min past head block time. We want to limit how long of an expiration allowed. Currently that is 1hr 15 minutes from head block time.

Reply:
Thanks for the explanation! Can we rename `expire` to something like: `latest_acceptable_expiration_time`?

Reply:
https://github.com/AntelopeIO/spring/pull/1503/commits/771ead5d583e543326fde6a38410425e67a01aaa